### ISSUE TITLE
How to avoid overfitting?

### ISSUE THREAD

**Issue by zzydog1982** [JsFlo/EmotionRecTraining#1]
Hi. I'm interested in FER. In your experiment, do you split the dataset into training/validation/testing set? How to avoid overfitting? Thanks!

**Comment by JsFlo** [comment]
Hi, The data actually comes labeled with "test" and "training" so I use those two sets to train and validate.
There are a couple things you could do to prevent overfitting.

 The first thing I would try is to bump the drop out or add more drop out (There is a drop out applied to the last dense/fc layers). 

Next, I would try stopping the training a little earlier. You could log the "training error" and the "test error" during training and pick out the place (n_epochs) where the test/validation set has a low error.

There are other techniques like gathering more data or changing the architecture that would be slightly more difficult to accomplish. This article goes over some of those:  https://elitedatascience.com/overfitting-in-machine-learning

**Comment by zzydog1982** [comment]
Thanks a lot. The data in the .csv are labeled with "private test" and "public test". If I can use the date with "private test " as my validation set, and use the other as test set?

**Comment by JsFlo** [comment]
Yea I used both "public/private test" as my validation set but you should probably separate them like that if you're going to be trying to reduce overfitting/modifying hyperparameters
